src/main/java/ru/ayaz/ChatServer.java

```diff
@@ -0,0 +1,184 @@
+package ru.ayaz;
+
+
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.io.PrintWriter;
+import java.net.ServerSocket;
+import java.net.Socket;
+import java.util.HashMap;
+import java.util.Map;
+
+public class ChatServer {
+
+    private ServerSocket serverSocket;
+    private Map<String, User> usersMap;
+    private Map<String, PrintWriter> outputWriterMap;
+
+
+    public ChatServer() {
+        try {
+            this.serverSocket = new ServerSocket(4400);
+            this.usersMap = new HashMap<String, User>();
+            this.outputWriterMap = new HashMap<String, PrintWriter>();
+        } catch (IOException e) {
+            e.printStackTrace();
+        }
+    }
+
+
+    public void start() {
+        try {
+            while (true) {
+                Socket socket = serverSocket.accept();
+                Thread t = new Thread(new UserHandler(socket));
+                t.start();
+            }
+        } catch (IOException e) {
+            e.printStackTrace();
+        }
+    }
+
+
+    private void sendMessageToEveryone(String message, String sender) {
+
+        System.out.println(message);
+
+        for (Map.Entry<String, User> entry : usersMap.entrySet()) {
+            String nickname = entry.getKey();
+            User user = entry.getValue();
+
+            if (user.isItMe(sender)) {
+                continue;
+            }
+
+            if (!user.isIgnored(sender)) {
+                PrintWriter pw = outputWriterMap.get(nickname);
+                pw.println(message);
+                pw.flush();
+            }
+
+        }
+
+    }
+
+
+    public class UserHandler implements Runnable {
```

Comment:
The way I see it is as follows:
There is a ChatServer which creates UserHandlers for each new connection. Each such user handler is registered with ChatServer (ChatServer  keeps track of UserHandlers). UserHandler's task is to listen for its user input and act on it. User types a command into his console, UserHandler performs certain actions to process it. Among those actions is broadcasting received message. But it doesn't broadcast received message on its own, it doesn't know where to find other users. What it does know, though, is that he is connected to the ChatServer, and ChatServer knows about every user connected to it. So, in order to broadcast a message to all users UserHandler delegates this command to the ChatServer.